File: load_all_diaries_and_comments.c
#include <stdio.h>
#include <string.h>
#include <dirent.h>

#define DIR_PATH "C:/Users/IWMAI/OneDrive/Personal-Diaries/"

char *read_single_file(char *dir_name, char *file_name)
{
    // load file name
    char df_name[256];
    strcpy(df_name, DIR_PATH);
    strcat(df_name, dir_name);
    strcat(df_name, file_name);
    // Read the content
    static char ans[8192];
    memset(ans, 0, sizeof(ans));
    FILE *fp;
    fp = fopen(df_name, "r");
    if (fp == NULL) {
        printf("Failed to open file %s\n", df_name);
        return ans;
        // exit(-1);
    }
    char s[100];
    while (1)
    {
        fscanf(fp, "%99s", s);
        strcat(ans, s);
        if (feof(fp))
            break;
    }
    return ans;
}

int main()
/*Write all the diaries and comments into a single file*/
{
    DIR *d;
    struct dirent *dir;
    d = opendir(DIR_PATH); // Opens the current directory
    FILE *fp;
    fp = fopen("test.txt", "w");

    if (d)
    {
        while ((dir = readdir(d)) != NULL)
        {
            fprintf(fp, "%s\n", read_single_file(dir->d_name, "/diary.txt"));
            // we do not guarantee that there is a comment file for each diary
            if (read_single_file(dir->d_name, "/comment.txt")[0] != '\0') continue;
            fprintf(fp, "%s\n\n", read_single_file(dir->d_name, "/comment.txt"));
        }
        closedir(d);
    } else {
        printf("Failed to open directory\n");
        return -1;
    }
    fclose(fp);

    return 0;
}

File: load_all_diaries_and_comments.py
print("starting to load diaries...")
import os
from config import diary_dir
import utils


def get_time_str(dir_name):
    y, mon, d, h, m, s = [int(i) for i in dir_name.split('-')]
    return f"(date: {y}.{mon}.{d}, time: {h}:{m})"


def load_all():
    diary_list = utils.load_all_dir_names(diary_dir)
    with open("all_diaries_and_comments.txt", "w", encoding="utf-8") as file:
        for dir_name in diary_list:
            print(f"loading {dir_name}")
            # for each dir, read the diary
            file.write(get_time_str(dir_name) + '\n')
            diary = utils.read_diary(os.path.join(diary_dir, dir_name))
            # diary = utils.remove_blank_lines(diary)
            file.write(diary + '\n\n')
            # read the comment
            if not os.path.exists(os.path.join(diary_dir, dir_name, "comment.txt")):
                continue
            comment = utils.read_comment(os.path.join(diary_dir, dir_name))
            # comment = utils.remove_blank_lines(comment)
            file.write(comment + '\n\n\n')


if __name__ == '__main__':
    # import time
    # start = time.time()
    load_all()
    # print(f"Time: {time.time()-start}s")
